exercises/slot_machine.livemd

```diff
@@ -0,0 +1,77 @@
+# Slot Machine
+
+## Navigation
+
+[Return Home](../start.livemd)<span style="padding: 0 30px"></span>
+[Report An Issue](https://github.com/DockYard-Academy/beta_curriculum/issues/new?assignees=&labels=&template=issue.md&title=)
+
+## The House Inc
+
+You have been hired at The House Inc.
+
+The House Inc is a casino right in the middle of Generic City. You aren't sure why a casino needs a programmer but you were desperate and applied anyway.
+
+Somehow, you actually made it past the other applicants and got the job... Or maybe that was because you were the only applicant.
+
+Your first assignment was to write the code for a slot machine. You looked at the specification and you were alarmed at how unfair this slot machine is. The House Inc isn't known for being a good place to work but you decide to talk to your boss about it anyway.
+
+So you went to your boss's office to voice your opinions on the topic and after you give a passionate speech on the value of ethics and morals. Your boss interrupts you in the middle of your ~~tantrum~~ speech and asks what ethics means.
+
+At that very moment you knew that there was no way these guys would listen because the only ethics they believe in are profit maximization.
+
+## Slot Machine Specs
+
+The slot machine is a `GenServer` that has the state of `balance` within it. Balance is how much money the machine has.
+
+* Slot machine reward sizes are random but must not exceed `balance`.
+* When a slot machine gives out a reward. Subtract `balance` by the amount of the reward.
+* The machine should only give rewards 30% of the time.
+
+This exercise will need you to use if statements. You will also need to figure out a way to generate random numbers and convert those into chances.
+
+```elixir
+defmodule SlotMachine do
+  use GenServer
+
+  def start() do
+    GenServer.start(__MODULE__, :ok)
+  end
+
+  def init(:ok) do
+    {:ok, %{balance: 7000}}
+  end
+
+  @doc """
+  Use this function for debugging. You need to make sure that your slot machine works properly.
```

Comment:
In the Client API, i'd like to add some documentation tests that will both help students understand the exercise, and in the future will be automatically run by livebook to provide test feedback.

For example

```elixir
@doc """
Retrieve the current balance of the slot machine.
The slot machine should start with 7000 pennies.

## Examples
        iex> {:ok, pid} = SlotMachine.start_link([])
        iex> SlotMachine.get_balance(pid)
        7000
"""

```

Reply:
I removed that function since you should be using `:sys.get_state` in this situation.

The question is: How do you write doctests that aren't confusing for a function that has random outputs?

Reply:
Why should you use `:sys.get_state`? That's usually only for debugging purposes. I personally like having a getter function.

For the randomness, see my comment below.

Reply:
My original intent for putting that getter function there was for debugging purposes. The idiomatic approach is to use :sys.get_state and thus, the change.